plugins/robots/common/ev3Kit/src/communication/ev3RobotCommunicationThread.cpp

```diff
@@ -49,9 +49,16 @@ QString Ev3RobotCommunicationThread::uploadFile(const QString &sourceFile, const
 	const QFileInfo fileInfo(sourceFile);
 	// A path to file on the remote device.
 	const QString devicePath = targetDir + "/" + fileInfo.fileName();
+
+	if (fileInfo.fileName().length() > mMaxFilenameLength) {
+		emit errorOccured(tr("Attempt to upload program with long filename. Please, shorten it to %1 symbols.")
```

Comment:
EV3 limits filename length to %1 characters, but you have %2, please, rename your project.